net/libfab/libfab.c

```diff
@@ -118,60 +125,219 @@ M0_INTERNAL void m0_net_libfab_fini(void)
 	*/
 }
 
+static void buf_complete(struct m0_fab__buf *buf, int32_t status);
+static void buf_done(struct m0_fab__buf *buf, int rc);
+static bool ma_invariant(const struct m0_fab__tm *ftm);
+static bool ma_is_locked(const struct m0_fab__tm *ftm);
+
+/**
+ * Helper function that posts a tm state change event.
+ */
+static void tm_event_post(struct m0_fab__tm *ftm, enum m0_net_tm_state state)
+{
+	struct m0_net_end_point *listen;
+
+	if (state == M0_NET_TM_STARTED) {
+		// Check for LISTENING Passive endpoint
+		listen = m0_tl_find(m0_nep, ne,
```

Comment:
No need to traverse the list as the listening passive endpoint is directly available from transfer_machine struct as _**tm->ftm_pep->fep_nep**_